The changelog says they unpinned it in v0.5.8:
  https://github.com/NVIDIA/TensorRT/blob/release/10.11/tools/onnx-graphsurgeon/CHANGELOG.md

And the bug that caused them to pin it has been closed:
  https://github.com/onnx/onnx/issues/6267

So I assume this just didn't make the code dump to GH.

diff --git a/tools/onnx-graphsurgeon/setup.py b/tools/onnx-graphsurgeon/setup.py
index a0ab7bf6..0ed75ba7 100644
--- a/tools/onnx-graphsurgeon/setup.py
+++ b/tools/onnx-graphsurgeon/setup.py
@@ -30,7 +30,7 @@ def no_publish():
 
 REQUIRED_PACKAGES = [
     "numpy",
-    "onnx>=1.14.0,<=1.16.1",
+    "onnx>=1.14.0",
 ]
 
 
